### ISSUE TITLE
Windows Docker Setup

### ISSUE THREAD

**Issue by veldyn** [veldyn/b2c-sysadmin-2024#9]

**Comment by veldyn** [comment]
1/26

I setup a windows VM at the end of Monday and planned on installing Docker on there today. This would allow me to experiment with Docker without potentially harming the integrity of my own machine, as well as to encounter any potential VM roadblocks. However, after installing Docker Desktop, it became clear that the windows version of docker is not nearly as well maintained. 

In order to make the installation process easier and decrease the level of work required to maintain this project in the future, we've decided to just use Ubuntu for Docker. For windows integration we can utilize WSL2 to run docker. This will cut down on instruction material significantly, since we can simply utilize the CLI for all of our instructions and run it all in the same environment. If WSL or windows compatibility changes later, or the UI for Docker Desktop changes, we won't need to re-do the instruction materials to account for these changes. Instead, we can change only the instructions for installing / configuring WSL2.

I'm looking at what will be necessary to configure for Windows systems right now to run this without issue, but we still need to discuss if we want to include and introductory instruction lab for these kinds of things, or if there are few enough of them that we can incorporate them into Lab1 without issue.

---

**Comment by veldyn** [comment]
It turns out we can actually install WSL easily through the Microsoft Store, so I've gone ahead and done that on my personal machine to install Docker. We also started writing the Lab for Intro to Docker today, and confirmed that all the same installation commands work on WSL that work on Ubuntu. I got Ubuntu up and running through WSL, and have written up that specific portion of the lab as well so we should be able to move forward simply by using Ubuntu from here on out.

We still need to finish up some Investigation into Docker to see what topics we want to cover exactly in the labs, but we can likely complete the installation lab without having to discuss these quite yet. We still have the general roadmap to follow for now.

We still have portions of Lab1 that need to be completed, such as the topic of "What is a Container?", elaboration on the commands utilized, as well as noting any configuration changes that must be made post-installation.
